lab/environments.py

```diff
@@ -425,7 +430,8 @@ def _get_job_params(self, step, is_last):
         job_params["time_limit_per_task"] = self.time_limit_per_task
         job_params["memory_per_cpu"] = self.memory_per_cpu
         memory_per_cpu_kb = SlurmEnvironment._get_memory_in_kb(self.memory_per_cpu)
-        job_params["soft_memory_limit"] = int(memory_per_cpu_kb * 0.98)
+        multiplier = self.cpus_per_task or 1
+        job_params["soft_memory_limit"] = int(multiplier * memory_per_cpu_kb * 0.98)
```

Comment:
When self.cpus_per_task is already defined above, we don't need the `multiplier` variable anymore and can use self.cpus_per_task directly.